src/dynamodb/client.py

```diff
@@ -176,6 +193,72 @@ def _create_client():
         )
 
 
+class S3Client(object):
+    """
+    Encapsulates S3 client interface, as exposed to the world. There is a single (singleton) instance of
+    S3Client in the process, which is lazily created upon the first request.
+    """
+
+    # the singleton instance of S3Client
+    _singleton = None
+
+    def __init__(self):
+        self.s3_client = S3Client._create_s3_client()
+        (
+            self.github_user_mapping_bucket_name,
+            self.github_user_mapping_key_name,
+        ) = GITHUB_USERNAMES_TO_ASANA_GIDS_S3_PATH.split("/", 1)
```

Comment:
providing this path is needed for the lambda to run then right since this assignment will fail unless we get a list of length 2?

Can you update the `main.tf` and `config.py` files to also specify that it is required and to not default to `""`? Or conversely, add error handling here?

Reply:
ah good point! yes, I'm just going to have this pathway 'fail' on the env var not being set, so that we revert to 'old pathway' if the env var isn't set. (I went with the anti-pattern of an `except Exception` ) in `get_asana_domain_user_id_from_github_handle` intentionally. 

In #166 I'll make sure that deployment fails if the env var isn't set.

FYI I also updated the instructions here: https://app.asana.com/0/1149418478823393/815237203241118/f to include the new env var.